src/pages/home/report/ReportActionsList.tsx

```diff
@@ -490,21 +492,37 @@ function ReportActionsList({
         return ReportUtils.isExpenseReport(report) || ReportUtils.isIOUReport(report);
     }, [parentReportAction, report, sortedVisibleReportActions]);
 
+    // storing the last read time used to render the unread marker
+    const markerLastReadTimeRef = useRef<string | undefined>();
```

Comment:
Do we really _need_ to save the `lastReadTime` in a ref twice? It is already being saved here:

https://github.com/Expensify/App/blob/b7f75efb4686cb6677a286f096af7988b0442765/src/pages/home/report/ReportActionsList.tsx#L198

Reply:
> so in case we re-calculate the unread marker, the lastReadTimeRef has been updated with report.lastReadTime value, causing the unread marker to disappear.

Where exactly is this happening? Is there a way we can prevent that from happening? From what I understand, the marker time should only change when the user marks a message as unread. So any other change would be unintended. 

Let's try fixing the original `lastReadTimeRef ` before adding another one since this functionality is already confusing and complex as it is.

Reply:
To be clear, I hired you for the job but your proposal was not fully satisfactory. Please work with me to get this PR in a better place.

I'm asking these questions again, please answer them directly: When the message is deleted, where _exactly_ is the `lastReadTimeRef` being cleared? Is there a way we can prevent that from happening? 

If the answer is yes (I believe it is), then we shouldn't need to save the time for the marker twice. Instead we can just fix the issue where `lastReadTimeRef` is being cleared when it shouldn't be.

Reply:
Thanks for the explanation. I took a closer look at this component and realized it needs a bit of a refactor to fix this issue properly and I can't really communicate that through PR comments. I'm going to merge your changes since they work and are covered with tests but I'm also going to make a follow up issue to clean this up.